From 741ab6c2483aac71251c50e0d616b018d921f0c2 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Fri, 20 Mar 2020 12:53:20 +0200
Subject: [PATCH] [SW-10603] goya2: RoT recovery progress update

Add to root of trust loading recovery from write protected flash
progress update to indicate that recovery image was loaded.

Change-Id: I577d6e31efb00fe2f41683c50f05d6586e6c8c80
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 1fd5be4..cc6b497 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -192,10 +192,13 @@
 {
 	uint32_t recovery_img_size = 0;
 
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_RECOVERY);
+
 	recovery_img_size = spi_flash_readl(0);
 	dma_copy_image(recovery_img_size, 0,
 		HL_SRAM_BASE + HL_ICCM_OFFSET, SPI_DMA_DST_HBW);
 	dma_wait_finish();
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_RECOVERY_DMA);
 
 	/* jump to iccm start location */
 	goto *(void *)(IMAGE_SIZE_LEN);
diff --git a/include/arcboot.h b/include/arcboot.h
index 8ac067d..ea176aa 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -105,7 +105,9 @@
 	PROGRESS_BRINGUP_RAM_SEC,
 	PROGRESS_BRINGUP_RAM_SEC_DMA,		/* 25 */
 	PROGRESS_BRINGUP_RAM_SEC_SECURITY,
-	PROGRESS_ARCBOOT_RAM_START
+	PROGRESS_ARCBOOT_RAM_START,
+	PROGRESS_ARCBOOT_RECOVERY,
+	PROGRESS_ARCBOOT_RECOVERY_DMA
 };
 
 enum security_location {
